code/modules/antagonists/bloodsucker/powers/bs_veil.dm

```diff
@@ -0,0 +1,165 @@
+
+
+
+
+
+/datum/action/bloodsucker/veil
+	name = "Veil of Many Faces"
+	desc = "Disguise yourself in the illusion of another identity."
+	button_icon_state = "power_veil"
+	bloodcost = 15
+	cooldown = 100
+	amToggle = TRUE
+	bloodsucker_can_buy = TRUE
+	warn_constant_cost = TRUE
+
+	// Outfit Vars
+	var/list/original_items = list()
+
+	// Identity Vars
+	var/prev_gender
+	var/prev_skin_tone
+	var/prev_hair_style
+	var/prev_facial_hair_style
+	var/prev_hair_color
+	var/prev_facial_hair_color
+	var/prev_underwear
+	var/prev_undershirt
+	var/prev_socks
```

Comment:
I suppose you'd could also store these undergarments' colors into separate vars, since that's a feature here.

Reply:
Ill do this tommorow.

Reply:
Done